src/Compression/CompressedReadBufferFromFile.cpp

```diff
@@ -17,34 +18,54 @@ namespace ErrorCodes
 
 bool CompressedReadBufferFromFile::nextImpl()
 {
-    size_t size_decompressed = 0;
-    size_t size_compressed_without_checksum;
-    size_compressed = readCompressedData(size_decompressed, size_compressed_without_checksum, false);
-    if (!size_compressed)
-        return false;
+    bool compressed_data_read_ok = false;
+    try
+    {
+        size_t size_decompressed = 0;
+        size_t size_compressed_without_checksum = 0;
+        size_compressed = readCompressedData(size_decompressed, size_compressed_without_checksum, false);
+        if (!size_compressed)
+            return false;
 
-    auto additional_size_at_the_end_of_buffer = codec->getAdditionalSizeAtTheEndOfBuffer();
+        compressed_data_read_ok = true;
+        LOG_TEST(log, "Decompressing {} bytes from {} to {} bytes", size_compressed, file_in.getFileName(), size_decompressed);
```

Comment:
This (and few other `LOG_TEST`) looks like way too verbose, @vitlibar  are you sure about this?

Reply:
BTW when I was debugging similar problem I wanted to get the payload itself, so, maybe instead of `LOG_TEST` we can add entry to the `system.crash_log` (or even create a new system table for such issues), that will contain the full payload. What do you think?